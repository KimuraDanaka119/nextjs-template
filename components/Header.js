import React from 'react';
import path from 'path';

import config from '../configs/config';

const Header = () => (
  <div className="header">
    <div className="container">
      <img src={path.join(config.paths.images, 'favicon.png')} />
      <span>This is header</span>
    </div>
  </div>
);

export default Header;
