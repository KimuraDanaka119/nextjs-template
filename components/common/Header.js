import React from 'react';
import PropTypes from 'prop-types';
import urljoin from 'url-join';
import Link from 'next/link';

import withI18N from '../hoc/withI18N';
import withConfig from '../hoc/withConfig';

const Header = ({ t, config }) => (
  <div className="header">
    <div className="container">
      <Link href="/index" as="/">
        <a><img src={urljoin(config.paths.images, 'favicon.png')} /></a>
      </Link>
      <span>{t('common:header.title')}</span>
    </div>
  </div>
);

Header.propTypes = {
  t: PropTypes.func,
  config: PropTypes.object,
};

export default withI18N(withConfig(Header));
export const undecorated = Header;
