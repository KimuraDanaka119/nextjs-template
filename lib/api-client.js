import fetch from 'isomorphic-unfetch';
import config from 'configs/config';

export default class ApiClient {
  async doFetch(path) {
    const url = config.api.baseURL + path;

    const res = await fetch(url);
    const data = await res.json();

    return {
      statusCode: res.status,
      data,
    };
  }

  async getShows(keyword) {
    const res = await this.doFetch(`/search/shows?q=${keyword}`);

    return res;
  }

  async getShow(id) {
    const res = await this.doFetch(`/shows/${id}`);

    return res;
  }
}
