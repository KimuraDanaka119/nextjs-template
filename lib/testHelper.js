import ApiClient from './api-client';

function t(key) {
  return key;
}

const config = {
  env: 'test',
  versionHash: 'test',
  paths: {
    static: '/static/test',
    styles: '/static/test/styles',
    images: '/static/test/images',
    locales: '/static/test/locales',
  },
  api: {
    baseURL: '.',
  },
};

const apiClient = new ApiClient('.');

export { t, config, apiClient };
