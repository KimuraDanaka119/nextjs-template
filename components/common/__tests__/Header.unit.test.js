import React from 'react';
import toJson from 'enzyme-to-json';
import { shallow } from 'enzyme';

import { undecorated as Header } from '../Header';
import { t, config } from '../../../lib/testHelper';

describe('<Header />', () => {
  it('should render correctly with imagePath=/', () => {
    const header = shallow(<Header t={t} config={config}/>);
    expect(toJson(header)).toMatchSnapshot();
  });
});
