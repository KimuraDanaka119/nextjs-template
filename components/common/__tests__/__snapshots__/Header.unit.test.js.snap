// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> should render correctly with imagePath=/ 1`] = `
<div
  className="header"
>
  <div
    className="container"
  >
    <Link
      as="/"
      href="/index"
    >
      <a>
        <img
          src="/static/test/images/favicon.png"
        />
      </a>
    </Link>
    <span>
      common:header.title
    </span>
  </div>
</div>
`;
